codec/decoder/core/src/decode_slice.cpp

```diff
@@ -318,7 +334,10 @@ int32_t WelsTargetMbConstruction (PWelsDecoderContext pCtx) {
     WelsMbIntraPredictionConstruction (pCtx, pCurLayer, 1);
   } else if (IS_INTER (pCurLayer->pMbType[pCurLayer->iMbXyIndex])) { //InterMB
     if (0 == pCurLayer->pCbp[pCurLayer->iMbXyIndex]) { //uiCbp==0 include SKIP
-      WelsMbInterPrediction (pCtx, pCurLayer);
+      if (!CheckRefPics (pCtx)) {
```

Comment:
Is this function necessary or too late to check here? Suppose it should be checked in parse process rather than in construction process.

Reply:
Not quite sure. It is a strange test case. With P Slice, pCtx->sRefPic.uiRefCount > 0 while pCtx->pRefPic.pRefList[LIST_0] is empty.

Reply:
It is weird that pCtx->pRefPic.pRefList[LIST_1][0] is not null (but uiRefCount[LIST_1] = 0) while pCtx->pRefPic.pRefList[LIST_0][0] is null (uiRefCount[LIST_0] = 1) for the P Slice for this testcase.

Reply:
For me, it seems not a perfect solution here. I think there should be somewhere to place better changes. I can agree for now. But need dig further.